src/sdi-refresh-monitor.c

```diff
@@ -545,6 +548,7 @@ static void sdi_refresh_monitor_dispose(GObject *object) {
   }
   g_clear_pointer(&self->snaps, g_hash_table_unref);
   g_clear_object(&self->client);
+  g_clear_pointer(&self->changes, g_hash_table_unref);
```

Comment:
You probably should be calling g_source_remove on each value, which might be doable by setting a function in g_hash_table_new_full below. This is an existing bug, but is solvable now you are storing the ID.

Reply:
In fact, since the timeout callback is called with a count_ref pointer to "self", this can be called only when there are no pending timeouts. Also, since I have to remove the change in the timeout callback, in that moment the source is already removed, so using new_full would produce an error (or, at least, a warning).

Reply:
I can add a comment to explain this, if you want.